core/types/calls.cc

```diff
@@ -372,25 +379,27 @@ MethodRef guessOverload(const GlobalState &gs, ClassOrModuleRef inClass, MethodR
             if (!overload.exists()) {
                 Exception::raise("Corruption of overloads?");
             } else {
-                allCandidates.emplace_back(overload);
+                allCandidates.emplace_back(i, overload);
                 current = overload;
             }
         }
 
-        fast_sort(allCandidates, [&](MethodRef s1, MethodRef s2) -> bool {
-            if (getArity(gs, s1) < getArity(gs, s2)) {
+        fast_sort(allCandidates, [&](const auto &s1, const auto &s2) -> bool {
+            const auto &[s1_idx, s1_sym] = s1;
+            const auto &[s2_idx, s2_sym] = s2;
+            if (getArity(gs, s1_sym) < getArity(gs, s2_sym)) {
                 return true;
             }
-            if (getArity(gs, s1) == getArity(gs, s2)) {
-                return s1.id() < s2.id();
+            if (getArity(gs, s1_sym) == getArity(gs, s2_sym)) {
```

Comment:
I guess it doesn't matter that we call `getArity` an extra time, since both the number of overloads and the arguments list should be small?